spanner/api/Spanner.Samples/UpdateDatabaseWithDefaultLeaderAsync.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_update_database_with_default_leader]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Threading.Tasks;
+
+public class UpdateDatabaseWithDefaultLeaderAsyncSample
+{
+    public async Task UpdateDatabaseWithDefaultLeaderAsync(string projectId, string instanceId, string databaseId, string defaultLeader)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+
+        using var connection = new SpannerConnection(connectionString);
+        var alterDatabaseStatement = @$"ALTER DATABASE `{databaseId}` SET OPTIONS
+                   (default_leader = '{defaultLeader}')";
+
+        using var alterDbCommand = connection.CreateDdlCommand(alterDatabaseStatement);
+
+        await alterDbCommand.ExecuteNonQueryAsync();
+
+        Console.WriteLine("Updated default leader");
```

Comment:
I think we want to do this with `UpdateDatabaseDdl` in the database admin client rather than through the data client.